src/praisonai/praisonai/bots/_session.py

```
@@ -199,26 +273,109 @@ def reap_stale(self, max_age_seconds: int) -> int:
+                return True
+                
+            # There's an event loop - we need to coordinate with asyncio locks
+            # This is more complex but necessary for thread safety
+            future = asyncio.run_coroutine_threadsafe(
+                self._add_mirror_entry_async(user_id, entry), loop
+            )
+            return future.result(timeout=10.0)  # 10 second timeout
+        
+        try:
+            return _sync_add_entry()
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)